receiver/snowflakereceiver/scraper.go

```diff
@@ -13,3 +13,197 @@
 // limitations under the License.
 
 package snowflakereceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snowflakereceiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snowflakereceiver/internal/metadata"
+)
+
+type snowflakeMetricsScraper struct {
+	client   *snowflakeClient
+	settings component.TelemetrySettings
+	conf     *Config
+	mb       *metadata.MetricsBuilder
+}
+
+func newSnowflakeMetricsScraper(settings receiver.CreateSettings, conf *Config) *snowflakeMetricsScraper {
+	return &snowflakeMetricsScraper{
+		settings: settings.TelemetrySettings,
+		conf:     conf,
+		mb:       metadata.NewMetricsBuilder(conf.Metrics, settings),
+	}
+}
+
+// for use with receiver.scraperhelper
+func (s *snowflakeMetricsScraper) start(_ context.Context, _ component.Host) (err error) {
+	s.client, err = newDefaultClient(s.settings, *s.conf)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *snowflakeMetricsScraper) shutdown(_ context.Context) (err error) {
+	err = s.client.client.Close()
+	return err
```

Comment:
It's not always guaranteed that shutdown is called after the start succeeds. So we need an extra check here
```suggestion
func (s *snowflakeMetricsScraper) shutdown(_ context.Context) (error) {
	if s.client == nil {
		return nil
	}
	return s.client.client.Close()
```

Reply:
got it, makes sense to me!